lib/app/backend/PipelineWatcher.js

```diff
@@ -1,22 +1,26 @@
 const chokidar = require('chokidar')
 const path = require('path')
-const {PIPELINES_FOLDER} = require('../AppSettings')
+const {PIPELINES_FOLDER, TRUSTED_PIPELINES_FOLDER} = require('../AppSettings')
 
 class PipelineWatcher {
   constructor (options) {
-    this.pipelineFolder = process.env.APP_PATH ? path.join(process.env.APP_PATH, PIPELINES_FOLDER) : PIPELINES_FOLDER
+    this.appFolder = process.env.APP_PATH ? process.env.APP_PATH : '.'
     this.options = options || {}
     this.options.ignoreInitial = true
     this.options.useFsEvents = false
   }
 
   start (cb) {
-    this.watcher = chokidar.watch(path.join(this.pipelineFolder, '*.json'), this.options)
+    this.watcher = chokidar.watch(path.join(this.appFolder, '**', '*.json'), this.options)
```

Comment:
watch only (trusted)pipeline folders, there should be pattern for this in this case you don't need the filter in line 20